kong/runloop/plugins_iterator.lua

```diff
@@ -123,110 +163,176 @@ local function get_next(self)
     local consumer_id = consumer and consumer.id or nil
 
     local name   = plugin.name
-    local combos = self.plugins_plan.combos[name]
+    local combos = self.iterator.combos[name]
 
     local plugin_configuration
 
-    repeat
+    if not combos then
+      goto after_combos
```

Comment:
Not sure, but is it possible to do this with function like `self:find_configuration()`, with early returns instead of `gotos`?

Reply:
I considered that, but I figured this would be faster than that (one less function call). Given that this is hot code I didn't want to make it slower.

Reply:
I am pretty sure this extra func call will be meaningless. And if performance is a thing, the repeat might have been even faster. How did you figure this would be faster in any meaningfull way?

Reply:
thought above comment, I don't feel strongly about it. It was just something in lines of this pr to make it more structured and easier to understand.